File: frontend/types/assurance.ts
export interface AssuranceCheck { key: string; label: string; status: 'passed' | 'failed'; violations: number; detail: string }
export interface AssuranceBlocker { key: string; status: 'blocked'; detail: string }
export interface AssuranceStatus { status: 'framework_ready' | 'failed'; migration_head: string; reporting_contract_version: string; checks: AssuranceCheck[]; blockers: AssuranceBlocker[] }

File: backend/app/schemas/assurance.py
"""Phase 11 structural assurance status contracts."""

from pydantic import BaseModel


class AssuranceCheck(BaseModel):
    key: str
    label: str
    status: str
    violations: int
    detail: str


class AssuranceBlocker(BaseModel):
    key: str
    status: str
    detail: str


class AssuranceStatus(BaseModel):
    status: str
    migration_head: str
    reporting_contract_version: str
    checks: list[AssuranceCheck]
    blockers: list[AssuranceBlocker]
